tests/integration_tests/performance/test_network_tcp_throughput.py

```diff
@@ -84,137 +59,38 @@ def get(self, metric_name: str, statistic_name: str) -> dict:
         return None
 
 
-def produce_iperf_output(
-    basevm, guest_cmd_builder, current_avail_cpu, runtime, omit, load_factor, modes
-):
-    """Produce iperf raw output from server-client connection."""
-    # Check if we have enough CPUs to pin the servers on the host.
-    # The available CPUs are the total minus vcpus, vmm and API threads.
-    assert load_factor * basevm.vcpus_count < CpuMap.len() - basevm.vcpus_count - 2
-
-    # Start the servers.
-    for server_idx in range(load_factor * basevm.vcpus_count):
-        assigned_cpu = CpuMap(current_avail_cpu)
-        iperf_server = (
-            CmdBuilder(f"taskset --cpu-list {assigned_cpu}")
-            .with_arg(basevm.jailer.netns_cmd_prefix())
-            .with_arg(IPERF3)
-            .with_arg("-sD")
-            .with_arg("-p", f"{BASE_PORT + server_idx}")
-            .with_arg("-1")
-            .build()
-        )
-        run_cmd(iperf_server)
-        current_avail_cpu += 1
+class TCPIPerf3Test(IPerf3Test):
+    """IPerf3 runner for the TCP throughput performance test"""
 
-    # Wait for iperf3 server to start.
-    time.sleep(SERVER_STARTUP_TIME_SEC)
+    def __init__(self, microvm, mode, host_ip, payload_length):
+        self._host_ip = host_ip
+        self._payload_length = payload_length
 
-    # Start `vcpus` iperf3 clients. We can not use iperf3 parallel streams
-    # due to non deterministic results and lack of scaling.
-    def spawn_iperf_client(conn, client_idx, mode):
-        # Add the port where the iperf3 client is going to send/receive.
-        cmd = (
-            guest_cmd_builder.with_arg("-p", f"{BASE_PORT + client_idx}")
-            .with_arg(mode)
-            .build()
+        super().__init__(
+            microvm,
+            BASE_PORT,
+            RUNTIME_SEC,
+            WARMUP_SEC,
+            mode,
+            LOAD_FACTOR * microvm.vcpus_count,
         )
-        pinned_cmd = f"taskset --cpu-list {client_idx % basevm.vcpus_count}" f" {cmd}"
-        _, stdout, _ = conn.execute_command(pinned_cmd)
-        return stdout.read()
-
-    # Remove inaccurate readings from the workloads end.
-    cpu_load_runtime = runtime - 2
-    assert cpu_load_runtime > 0
-    with concurrent.futures.ThreadPoolExecutor() as executor:
-        futures = []
-        cpu_load_future = executor.submit(
-            get_cpu_percent, basevm.jailer_clone_pid, cpu_load_runtime, omit
+
+    def guest_command(self, port_offset):
+        cmd = (
+            super()
+            .guest_command(port_offset)
+            .with_arg("--client", self._host_ip)
```

Comment:
I think we can move this to the parent's `guest_command()`, since both iperf3 and iperf3-vsock commands can take either `--client <host>` or `-c <host>`:

https://iperf.fr/iperf-doc.php
> -c, --client host | Run iPerf in client mode, connecting to an iPerf server running on host.

https://github.com/stefano-garzarella/iperf-vsock/blob/master/docs/invoking.rst
>        -c, --client host
>              run  in  client  mode,  connecting  to the specified server.  By
>              default, a test consists of sending data from the client to  the
>              server, unless the -R flag is specified.

Reply:
hmm, I'd prefer to change it to `host` rather than `client`, since the `-c, --client` means turning on the client mode and the passed arg means "host".

Reply:
ah, yeah, you're right. What a confusingly named argument. Maybe `connect_to` for the instance variable?

Reply:
That looks great to me!